### ISSUE TITLE
systemd-udevd: Process '/sbin/crda' failed with exit code 234.

### ISSUE THREAD

**Issue by morfikov** [systemd/systemd#922]
I'm not sure but I think the message in question started to show after updating systemd to v223 or v224. I know for sure the problem didn't exist in v222.

When I sleep my laptop, everything looks just fine, but when I wake it up I can see the message in the journal. The problem is that systemd-udevd is looped:

```
Aug 10 04:02:26 morfikownia kernel: cfg80211: Calling CRDA to update world regulatory domain
Aug 10 04:02:26 morfikownia kernel: cfg80211: World regulatory domain updated:
Aug 10 04:02:26 morfikownia kernel: cfg80211:  DFS Master region: unset
Aug 10 04:02:26 morfikownia kernel: cfg80211:   (start_freq - end_freq @ bandwidth), (max_antenna_gain, max_eirp), (dfs_cac_time)
Aug 10 04:02:26 morfikownia kernel: cfg80211:   (2402000 KHz - 2472000 KHz @ 40000 KHz), (N/A, 2000 mBm), (N/A)
Aug 10 04:02:26 morfikownia kernel: cfg80211:   (2457000 KHz - 2482000 KHz @ 40000 KHz), (N/A, 2000 mBm), (N/A)
Aug 10 04:02:26 morfikownia kernel: cfg80211:   (2474000 KHz - 2494000 KHz @ 20000 KHz), (N/A, 2000 mBm), (N/A)
Aug 10 04:02:26 morfikownia kernel: cfg80211:   (5170000 KHz - 5250000 KHz @ 80000 KHz, 160000 KHz AUTO), (N/A, 2000 mBm), (N/A)
Aug 10 04:02:26 morfikownia kernel: cfg80211:   (5250000 KHz - 5330000 KHz @ 80000 KHz, 160000 KHz AUTO), (N/A, 2000 mBm), (0 s)
Aug 10 04:02:26 morfikownia kernel: cfg80211:   (5490000 KHz - 5730000 KHz @ 160000 KHz), (N/A, 2000 mBm), (0 s)
Aug 10 04:02:26 morfikownia kernel: cfg80211:   (5735000 KHz - 5835000 KHz @ 80000 KHz), (N/A, 2000 mBm), (N/A)
Aug 10 04:02:26 morfikownia kernel: cfg80211:   (57240000 KHz - 63720000 KHz @ 2160000 KHz), (N/A, 0 mBm), (N/A)
Aug 10 04:02:26 morfikownia kernel: cfg80211: Calling CRDA for country: 97
Aug 10 04:02:26 morfikownia systemd-udevd[65955]: Process '/sbin/crda' failed with exit code 234.

Aug 10 04:02:29 morfikownia kernel: cfg80211: Calling CRDA to update world regulatory domain
Aug 10 04:02:29 morfikownia kernel: cfg80211: World regulatory domain updated:
Aug 10 04:02:29 morfikownia kernel: cfg80211:  DFS Master region: unset
Aug 10 04:02:29 morfikownia kernel: cfg80211:   (start_freq - end_freq @ bandwidth), (max_antenna_gain, max_eirp), (dfs_cac_time)
Aug 10 04:02:29 morfikownia kernel: cfg80211:   (2402000 KHz - 2472000 KHz @ 40000 KHz), (N/A, 2000 mBm), (N/A)
Aug 10 04:02:29 morfikownia kernel: cfg80211:   (2457000 KHz - 2482000 KHz @ 40000 KHz), (N/A, 2000 mBm), (N/A)
Aug 10 04:02:29 morfikownia kernel: cfg80211:   (2474000 KHz - 2494000 KHz @ 20000 KHz), (N/A, 2000 mBm), (N/A)
Aug 10 04:02:29 morfikownia kernel: cfg80211:   (5170000 KHz - 5250000 KHz @ 80000 KHz, 160000 KHz AUTO), (N/A, 2000 mBm), (N/A)
Aug 10 04:02:29 morfikownia kernel: cfg80211:   (5250000 KHz - 5330000 KHz @ 80000 KHz, 160000 KHz AUTO), (N/A, 2000 mBm), (0 s)
Aug 10 04:02:29 morfikownia kernel: cfg80211:   (5490000 KHz - 5730000 KHz @ 160000 KHz), (N/A, 2000 mBm), (0 s)
Aug 10 04:02:29 morfikownia kernel: cfg80211:   (5735000 KHz - 5835000 KHz @ 80000 KHz), (N/A, 2000 mBm), (N/A)
Aug 10 04:02:29 morfikownia kernel: cfg80211:   (57240000 KHz - 63720000 KHz @ 2160000 KHz), (N/A, 0 mBm), (N/A)
Aug 10 04:02:29 morfikownia kernel: cfg80211: Calling CRDA for country: 97
Aug 10 04:02:29 morfikownia systemd-udevd[65974]: Process '/sbin/crda' failed with exit code 234.
...
```

It spams the message all the time in interval of 3-4 secs. The WiFi connection works.

The only way to make it stop is to unload the WiFi kernel module -- in this case `brcmsmac` . After inserting it again, the whole spam disappears, but there's another problem:

```
Aug 10 04:10:29 morfikownia kernel: Broadcom 43xx driver loaded [ Features: PMNLS ]
Aug 10 04:10:29 morfikownia kernel: cfg80211: Calling CRDA for country: PL
Aug 10 04:10:29 morfikownia systemd[1]: Starting Load/Save RF Kill Switch Status of rfkill2...
Aug 10 04:10:29 morfikownia kernel: cfg80211: Regulatory domain changed to country: PL
Aug 10 04:10:29 morfikownia kernel: cfg80211:  DFS Master region: ETSI
Aug 10 04:10:29 morfikownia kernel: cfg80211:   (start_freq - end_freq @ bandwidth), (max_antenna_gain, max_eirp), (dfs_cac_time)
Aug 10 04:10:29 morfikownia kernel: cfg80211:   (2402000 KHz - 2482000 KHz @ 40000 KHz), (N/A, 2000 mBm), (N/A)
Aug 10 04:10:29 morfikownia kernel: cfg80211:   (5170000 KHz - 5250000 KHz @ 80000 KHz, 160000 KHz AUTO), (N/A, 2000 mBm), (N/A)
Aug 10 04:10:29 morfikownia kernel: cfg80211:   (5250000 KHz - 5330000 KHz @ 80000 KHz, 160000 KHz AUTO), (N/A, 2000 mBm), (0 s)
Aug 10 04:10:29 morfikownia kernel: cfg80211:   (5490000 KHz - 5710000 KHz @ 160000 KHz), (N/A, 2700 mBm), (0 s)
Aug 10 04:10:29 morfikownia kernel: cfg80211:   (57000000 KHz - 66000000 KHz @ 2160000 KHz), (N/A, 4000 mBm), (N/A)
Aug 10 04:10:29 morfikownia systemd[1]: Started Load/Save RF Kill Switch Status of rfkill2.
Aug 10 04:10:30 morfikownia systemd-networkd[1479]: wlan0           : found matching network '/etc/systemd/network/60-bounded-wlan0.network', based on potentially unpredictable ifname
Aug 10 04:10:30 morfikownia systemd-networkd[1479]: wlan0: wlan0           : could not join netdev: Operation not permitted
Aug 10 04:10:30 morfikownia systemd-networkd[1479]: wlan0: Failed
Aug 10 04:10:30 morfikownia kernel: bond0: wlan0 is up - this may be due to an out of date ifenslave
```

Sometimes I have the same kind of message (`wlan0: could not join netdev: Operation not permitted`) in the boot phase, but usually after 30 sec - 1min I can restart `systemd-networkd` service and everything backs to normal, but not in this case. Here it simple doesn't work and `systemd-networkd` is unable to configure the WiFi interface. Here's the log that I can get when restarting systemd-networkd.

```
Aug 10 04:12:15 morfikownia polkitd(authority=local)[1594]: Registered Authentication Agent for unix-process:69616:5481560 (system bus name :1.65 [/usr/bin/pkttyagent --notify-fd 5 --fallback], object path /org/freedesktop/PolicyKit1/AuthenticationAgent, locale en_US.UTF-8)
Aug 10 04:12:15 morfikownia systemd[1]: lvm2.service: Cannot add dependency job, ignoring: Unit lvm2.service is masked.
Aug 10 04:12:15 morfikownia systemd[1]: Stopping WPA supplicant (wlan0)...
Aug 10 04:12:15 morfikownia kernel: wlan0: deauthenticating from c4:6e:1f:95:ef:fd by local choice (Reason: 3=DEAUTH_LEAVING)
Aug 10 04:12:15 morfikownia systemd-networkd[69553]: wlan0: Lost carrier
Aug 10 04:12:15 morfikownia kernel: brcmsmac bcma0:1: brcmsmac: brcms_ops_bss_info_changed: disassociated
Aug 10 04:12:15 morfikownia kernel: brcmsmac bcma0:1: brcms_ops_bss_info_changed: qos enabled: false (implement)
Aug 10 04:12:15 morfikownia kernel: cfg80211: Calling CRDA to update world regulatory domain
Aug 10 04:12:15 morfikownia kernel: cfg80211: World regulatory domain updated:
Aug 10 04:12:15 morfikownia kernel: cfg80211:  DFS Master region: unset
Aug 10 04:12:15 morfikownia kernel: cfg80211:   (start_freq - end_freq @ bandwidth), (max_antenna_gain, max_eirp), (dfs_cac_time)
Aug 10 04:12:15 morfikownia kernel: cfg80211:   (2402000 KHz - 2472000 KHz @ 40000 KHz), (N/A, 2000 mBm), (N/A)
Aug 10 04:12:15 morfikownia kernel: cfg80211:   (2457000 KHz - 2482000 KHz @ 40000 KHz), (N/A, 2000 mBm), (N/A)
Aug 10 04:12:15 morfikownia kernel: cfg80211:   (2474000 KHz - 2494000 KHz @ 20000 KHz), (N/A, 2000 mBm), (N/A)
Aug 10 04:12:15 morfikownia kernel: cfg80211:   (5170000 KHz - 5250000 KHz @ 80000 KHz, 160000 KHz AUTO), (N/A, 2000 mBm), (N/A)
Aug 10 04:12:15 morfikownia kernel: cfg80211:   (5250000 KHz - 5330000 KHz @ 80000 KHz, 160000 KHz AUTO), (N/A, 2000 mBm), (0 s)
Aug 10 04:12:15 morfikownia kernel: cfg80211:   (5490000 KHz - 5730000 KHz @ 160000 KHz), (N/A, 2000 mBm), (0 s)
Aug 10 04:12:15 morfikownia kernel: cfg80211:   (5735000 KHz - 5835000 KHz @ 80000 KHz), (N/A, 2000 mBm), (N/A)
Aug 10 04:12:15 morfikownia kernel: cfg80211:   (57240000 KHz - 63720000 KHz @ 2160000 KHz), (N/A, 0 mBm), (N/A)
Aug 10 04:12:15 morfikownia kernel: cfg80211: Calling CRDA for country: PL
Aug 10 04:12:15 morfikownia kernel: cfg80211: Regulatory domain changed to country: PL
Aug 10 04:12:15 morfikownia kernel: cfg80211:  DFS Master region: ETSI
Aug 10 04:12:15 morfikownia kernel: cfg80211:   (start_freq - end_freq @ bandwidth), (max_antenna_gain, max_eirp), (dfs_cac_time)
Aug 10 04:12:15 morfikownia kernel: cfg80211:   (2402000 KHz - 2482000 KHz @ 40000 KHz), (N/A, 2000 mBm), (N/A)
Aug 10 04:12:15 morfikownia kernel: cfg80211:   (5170000 KHz - 5250000 KHz @ 80000 KHz, 160000 KHz AUTO), (N/A, 2000 mBm), (N/A)
Aug 10 04:12:15 morfikownia kernel: cfg80211:   (5250000 KHz - 5330000 KHz @ 80000 KHz, 160000 KHz AUTO), (N/A, 2000 mBm), (0 s)
Aug 10 04:12:15 morfikownia kernel: cfg80211:   (5490000 KHz - 5710000 KHz @ 160000 KHz), (N/A, 2700 mBm), (0 s)
Aug 10 04:12:15 morfikownia kernel: cfg80211:   (57000000 KHz - 66000000 KHz @ 2160000 KHz), (N/A, 4000 mBm), (N/A)
Aug 10 04:12:15 morfikownia systemd[1]: Stopped WPA supplicant (wlan0).
Aug 10 04:12:15 morfikownia systemd[1]: Stopping Network Service...
Aug 10 04:12:15 morfikownia systemd[1]: Stopped Network Service.
Aug 10 04:12:15 morfikownia systemd[1]: Starting Network Service...
Aug 10 04:12:15 morfikownia systemd-networkd[69627]: bond0: netdev ready
Aug 10 04:12:15 morfikownia systemd-timesyncd[1166]: No network connectivity, watching for changes.
Aug 10 04:12:15 morfikownia systemd-timesyncd[1166]: No network connectivity, watching for changes.
Aug 10 04:12:15 morfikownia systemd-networkd[69627]: br-lxc: netdev ready
Aug 10 04:12:15 morfikownia systemd-timesyncd[1166]: No network connectivity, watching for changes.
Aug 10 04:12:15 morfikownia systemd-timesyncd[1166]: No network connectivity, watching for changes.
Aug 10 04:12:15 morfikownia systemd-timesyncd[1166]: No network connectivity, watching for changes.
Aug 10 04:12:15 morfikownia kernel: bond0: wlan0 is up - this may be due to an out of date ifenslave
Aug 10 04:12:15 morfikownia systemd-networkd[69627]: Enumeration completed
Aug 10 04:12:15 morfikownia systemd-networkd[69627]: br-lxc: netdev exists, using existing without changing its parameters
Aug 10 04:12:15 morfikownia systemd-networkd[69627]: bond0: netdev exists, using existing without changing its parameters
Aug 10 04:12:15 morfikownia systemd[1]: Started Network Service.
Aug 10 04:12:15 morfikownia systemd-networkd[69627]: wlan0           : found matching network '/etc/systemd/network/60-bounded-wlan0.network', based on potentially unpredictable ifname
Aug 10 04:12:15 morfikownia systemd[1]: Starting WPA supplicant (wlan0)...
Aug 10 04:12:15 morfikownia systemd-networkd[69627]: wlan0: wlan0           : could not join netdev: Operation not permitted
Aug 10 04:12:15 morfikownia systemd-networkd[69627]: wlan0: Failed
Aug 10 04:12:15 morfikownia systemd-networkd[69627]: br-lxc: Configured
Aug 10 04:12:15 morfikownia systemd-networkd[69627]: eth1: Configured
Aug 10 04:12:15 morfikownia polkitd(authority=local)[1594]: Unregistered Authentication Agent for unix-process:69616:5481560 (system bus name :1.65, object path /org/freedesktop/PolicyKit1/AuthenticationAgent, locale en_US.UTF-8) (disconnected from bus)
Aug 10 04:12:15 morfikownia systemd-networkd[69627]: lo: Configured
Aug 10 04:12:15 morfikownia systemd[1]: Started WPA supplicant (wlan0).
Aug 10 04:12:16 morfikownia kernel: wlan0: authenticate with c4:6e:1f:95:ef:fd
Aug 10 04:12:16 morfikownia kernel: wlan0: send auth to c4:6e:1f:95:ef:fd (try 1/3)
Aug 10 04:12:16 morfikownia kernel: wlan0: authenticated
Aug 10 04:12:16 morfikownia kernel: wlan0: associate with c4:6e:1f:95:ef:fd (try 1/3)
Aug 10 04:12:16 morfikownia kernel: wlan0: RX AssocResp from c4:6e:1f:95:ef:fd (capab=0x431 status=0 aid=1)
Aug 10 04:12:16 morfikownia kernel: brcmsmac bcma0:1: brcmsmac: brcms_ops_bss_info_changed: associated
Aug 10 04:12:16 morfikownia kernel: brcmsmac bcma0:1: brcms_ops_bss_info_changed: qos enabled: true (implement)
Aug 10 04:12:16 morfikownia kernel: wlan0: associated
Aug 10 04:12:16 morfikownia kernel: brcmsmac bcma0:1: wl0: brcms_c_d11hdrs_mac80211:  txop exceeded phylen 159/256 dur 1778/1504
Aug 10 04:12:16 morfikownia kernel: cfg80211: Calling CRDA for country: PL
Aug 10 04:12:16 morfikownia kernel: brcmsmac bcma0:1: wl0: brcms_c_d11hdrs_mac80211:  txop exceeded phylen 137/256 dur 1602/1504
Aug 10 04:12:16 morfikownia systemd-networkd[69627]: wlan0: Gained carrier
Aug 10 04:12:16 morfikownia kernel: cfg80211: Regulatory domain changed to country: PL
Aug 10 04:12:16 morfikownia kernel: cfg80211:  DFS Master region: ETSI
Aug 10 04:12:16 morfikownia kernel: cfg80211:   (start_freq - end_freq @ bandwidth), (max_antenna_gain, max_eirp), (dfs_cac_time)
Aug 10 04:12:16 morfikownia kernel: cfg80211:   (2402000 KHz - 2482000 KHz @ 40000 KHz), (N/A, 2000 mBm), (N/A)
Aug 10 04:12:16 morfikownia kernel: cfg80211:   (5170000 KHz - 5250000 KHz @ 80000 KHz, 160000 KHz AUTO), (N/A, 2000 mBm), (N/A)
Aug 10 04:12:16 morfikownia kernel: cfg80211:   (5250000 KHz - 5330000 KHz @ 80000 KHz, 160000 KHz AUTO), (N/A, 2000 mBm), (0 s)
Aug 10 04:12:16 morfikownia kernel: cfg80211:   (5490000 KHz - 5710000 KHz @ 160000 KHz), (N/A, 2700 mBm), (0 s)
Aug 10 04:12:16 morfikownia kernel: cfg80211:   (57000000 KHz - 66000000 KHz @ 2160000 KHz), (N/A, 4000 mBm), (N/A)
```

Even though it's authenticated and associated, it couldn't get the IP configuration via dhcp:

```
# networkctl list
IDX LINK             TYPE               OPERATIONAL SETUP
  1 lo               loopback           carrier     configured
  2 eth1             ether              no-carrier  configured
  3 ifb0             ether              degraded    unmanaged
  4 ifb1             ether              degraded    unmanaged
  6 br-lxc           ether              no-carrier  configured
  7 bond0            ether              routable    configuring
  8 wlan0            wlan               degraded    failed
```

I managed to fix this problem too. As you can see, there's something like `bond0: wlan0 is up - this may be due to an out of date ifenslave` . So I typed:

```
# ip link set wlan0 down
# systemctl restart systemd-networkd
```

And after that I finally got the WiFi connection:

```
Aug 10 04:15:46 morfikownia polkitd(authority=local)[1594]: Registered Authentication Agent for unix-process:70786:5502652 (system bus name :1.81 [/usr/bin/pkttyagent --notify-fd 5 --fallback], object path /org/freedesktop/PolicyKit1/AuthenticationAgent, locale en_US.UTF-8)
Aug 10 04:15:46 morfikownia systemd[1]: lvm2.service: Cannot add dependency job, ignoring: Unit lvm2.service is masked.
Aug 10 04:15:46 morfikownia systemd[1]: Stopping WPA supplicant (wlan0)...
Aug 10 04:15:46 morfikownia systemd[1]: Stopped WPA supplicant (wlan0).
Aug 10 04:15:46 morfikownia systemd[1]: Stopping Network Service...
Aug 10 04:15:46 morfikownia systemd[1]: Stopped Network Service.
Aug 10 04:15:46 morfikownia systemd[1]: Starting Network Service...
Aug 10 04:15:46 morfikownia systemd-timesyncd[1166]: No network connectivity, watching for changes.
Aug 10 04:15:46 morfikownia systemd-timesyncd[1166]: No network connectivity, watching for changes.
Aug 10 04:15:46 morfikownia systemd-networkd[70795]: bond0: netdev ready
Aug 10 04:15:46 morfikownia systemd-timesyncd[1166]: No network connectivity, watching for changes.
Aug 10 04:15:46 morfikownia systemd-networkd[70795]: br-lxc: netdev ready
Aug 10 04:15:46 morfikownia systemd-timesyncd[1166]: No network connectivity, watching for changes.
Aug 10 04:15:46 morfikownia systemd-timesyncd[1166]: No network connectivity, watching for changes.
Aug 10 04:15:46 morfikownia systemd-timesyncd[1166]: No network connectivity, watching for changes.
Aug 10 04:15:46 morfikownia systemd-timesyncd[1166]: No network connectivity, watching for changes.
Aug 10 04:15:46 morfikownia systemd-networkd[70795]: Enumeration completed
Aug 10 04:15:46 morfikownia systemd-networkd[70795]: br-lxc: netdev exists, using existing without changing its parameters
Aug 10 04:15:46 morfikownia systemd-networkd[70795]: bond0: netdev exists, using existing without changing its parameters
Aug 10 04:15:46 morfikownia systemd[1]: Started Network Service.
Aug 10 04:15:46 morfikownia systemd[1]: Starting WPA supplicant (wlan0)...
Aug 10 04:15:46 morfikownia systemd-networkd[70795]: wlan0           : found matching network '/etc/systemd/network/60-bounded-wlan0.network', based on potentially unpredictable ifname
Aug 10 04:15:46 morfikownia polkitd(authority=local)[1594]: Unregistered Authentication Agent for unix-process:70786:5502652 (system bus name :1.81, object path /org/freedesktop/PolicyKit1/AuthenticationAgent, locale en_US.UTF-8) (disconnected from bus)
Aug 10 04:15:46 morfikownia kernel: brcmsmac bcma0:1: brcms_ops_bss_info_changed: qos enabled: false (implement)
Aug 10 04:15:46 morfikownia kernel: brcmsmac bcma0:1: brcms_ops_config: change power-save mode: false (implement)
Aug 10 04:15:46 morfikownia kernel: IPv6: ADDRCONF(NETDEV_UP): wlan0: link is not ready
Aug 10 04:15:46 morfikownia kernel: bond0: Enslaving wlan0 as an active interface with a down link
Aug 10 04:15:46 morfikownia systemd-networkd[70795]: wlan0: Configured
Aug 10 04:15:46 morfikownia systemd-networkd[70795]: br-lxc: Configured
Aug 10 04:15:46 morfikownia systemd-networkd[70795]: eth1: Configured
Aug 10 04:15:46 morfikownia systemd-networkd[70795]: lo: Configured
Aug 10 04:15:46 morfikownia systemd-networkd[70795]: bond0: Lost carrier
Aug 10 04:15:46 morfikownia systemd-networkd[70795]: bond0: DHCPv4 connection considered system critical, ignoring request to reconfigure it.
Aug 10 04:15:46 morfikownia systemd[1]: Started WPA supplicant (wlan0).
Aug 10 04:15:47 morfikownia kernel: wlan0: authenticate with c4:6e:1f:95:ef:fd
Aug 10 04:15:47 morfikownia kernel: wlan0: send auth to c4:6e:1f:95:ef:fd (try 1/3)
Aug 10 04:15:47 morfikownia kernel: wlan0: authenticated
Aug 10 04:15:47 morfikownia kernel: wlan0: associate with c4:6e:1f:95:ef:fd (try 1/3)
Aug 10 04:15:47 morfikownia kernel: wlan0: RX AssocResp from c4:6e:1f:95:ef:fd (capab=0x431 status=0 aid=1)
Aug 10 04:15:47 morfikownia kernel: brcmsmac bcma0:1: brcmsmac: brcms_ops_bss_info_changed: associated
Aug 10 04:15:47 morfikownia kernel: brcmsmac bcma0:1: brcms_ops_bss_info_changed: qos enabled: true (implement)
Aug 10 04:15:47 morfikownia kernel: wlan0: associated
Aug 10 04:15:47 morfikownia kernel: cfg80211: Calling CRDA for country: PL
Aug 10 04:15:47 morfikownia kernel: cfg80211: Regulatory domain changed to country: PL
Aug 10 04:15:47 morfikownia kernel: cfg80211:  DFS Master region: ETSI
Aug 10 04:15:47 morfikownia kernel: cfg80211:   (start_freq - end_freq @ bandwidth), (max_antenna_gain, max_eirp), (dfs_cac_time)
Aug 10 04:15:47 morfikownia kernel: cfg80211:   (2402000 KHz - 2482000 KHz @ 40000 KHz), (N/A, 2000 mBm), (N/A)
Aug 10 04:15:47 morfikownia kernel: cfg80211:   (5170000 KHz - 5250000 KHz @ 80000 KHz, 160000 KHz AUTO), (N/A, 2000 mBm), (N/A)
Aug 10 04:15:47 morfikownia kernel: cfg80211:   (5250000 KHz - 5330000 KHz @ 80000 KHz, 160000 KHz AUTO), (N/A, 2000 mBm), (0 s)
Aug 10 04:15:47 morfikownia kernel: cfg80211:   (5490000 KHz - 5710000 KHz @ 160000 KHz), (N/A, 2700 mBm), (0 s)
Aug 10 04:15:47 morfikownia kernel: cfg80211:   (57000000 KHz - 66000000 KHz @ 2160000 KHz), (N/A, 4000 mBm), (N/A)
Aug 10 04:15:47 morfikownia kernel: brcmsmac bcma0:1: wl0: brcms_c_d11hdrs_mac80211:  txop exceeded phylen 159/256 dur 1778/1504
Aug 10 04:15:47 morfikownia kernel: brcmsmac bcma0:1: wl0: brcms_c_d11hdrs_mac80211:  txop exceeded phylen 137/256 dur 1602/1504
Aug 10 04:15:47 morfikownia systemd-networkd[70795]: wlan0: Gained carrier
Aug 10 04:15:47 morfikownia systemd-networkd[70795]: bond0: Gained carrier
Aug 10 04:15:47 morfikownia kernel: bond0: link status up for interface wlan0, enabling it in 0 ms
Aug 10 04:15:47 morfikownia kernel: bond0: link status definitely up for interface wlan0, 0 Mbps full duplex
Aug 10 04:15:47 morfikownia kernel: bond0: first active interface up!
Aug 10 04:15:47 morfikownia systemd-networkd[70795]: bond0: DHCPv4 address 192.168.1.150/24 via 192.168.1.1
Aug 10 04:15:47 morfikownia systemd-networkd[70795]: bond0: Configured
```

Does anyone have an idea why this kind of situations happen?

**Comment by poettering** [comment]
/sbin/crda is from the "crda" package, not from systemd. Please ask the crda and wlan kernel driver folks for help.

**Comment by poettering** [comment]
I think the crda issue might be because your wlan device sees one beacon advertising you are in country "PL" as well as another one advertising a country "97", and it keeps switching the world regulatory domain all the time due to this. And the crda tool probably fails because "97" is no valid country identifier... But that's just a guess, please ask crda/wlan people for help. (most likely some AP in your vicinity just advertises rubbish country information, and the crda/wlan code doesn't handle this nicely...)

**Comment by poettering** [comment]
And I am pretty sure the other issue you described is the same as #936. Closing hence.

**Comment by morfikov** [comment]
Actually that's my AP, and its configuration has been the same all the time. There's OpenWRT installed, and hostapd advertises `country_code=PL` . So I don't think the configuration of AP is involved, but the problem only appears after the laptop is suspended and then woken up. When I start the machine normally, there's no problem with the WiFi connection. But the strangest thing is that after reporting this issue, I haven't experienced it since then. Anyway thanks for the hint with the 97 non valid code.

**Comment by teg** [comment]
Sorry for being late to the party, but just a note: the (most likely) reason you are seeing warnings now that you did not use to is that I upgraded this log message from 'debug' to 'warning' level, so most likely crda was always failing, it just did not appear in the logs.

**Comment by morfikov** [comment]
I don't think the reason was the log level because the same messages I can observe now. The only problem was the loop with the  97 country code. The WiFi connection was working, so without looking into the journal I couldn't even tell if something was wrong, but when I saw thousands of this kind of messages... And as I said before, that only was happening when the laptop has woken up -- it didn't happen when I power off and on the machine. Maybe something is wrong with the AP, but why that doesn't happen now? Anyway I'll keep an eye on it and try to make some tests when the situation will occur again.

Actually there's two problems here, one with the loop, and the other concerning `wlan0  : could not join netdev: Operation not permitted` . I don't think both are connected because I reported the second problem [here](http://lists.freedesktop.org/archives/systemd-devel/2015-February/028869.html) (point 3) a long time ago.
